### PR TITLE
Supported json in message table.

### PR DESCRIPTION
**Pull request by dartartem**

### CODE REVIEW

File: eventuate-common-jdbc/src/main/java/io/eventuate/common/jdbc/sqldialect/EventuateSqlDialect.java [hunk 1]
```diff
@@ -2,6 +2,16 @@
...
   String getCurrentTimeInMillisecondsExpression();
+
   String addLimitToSql(String sql, String limitExpression);
+
+  default String castToJson(String sqlPart) {
+    return sqlPart;
+  }
+
+  default String objectToString(Object object) {
+    return object.toString();
```
Comment (cer) [inline eventuate-common-jdbc/src/main/java/io/eventuate/common/jdbc/sqldialect/EventuateSqlDialect.java:15]: Where is this called? No where else in this PR?
Also, accepting any object and calling toString doesn't seem right.

File: eventuate-common-jdbc/src/main/java/io/eventuate/common/jdbc/sqldialect/PostgresDialect.java [hunk 2]
```diff
@@ -15,4 +17,18 @@ public boolean supports(String driver) {
   public int getOrder() {
     return Integer.MIN_VALUE;
   }
+
+  @Override
+  public String castToJson(String sqlPart) {
+    return sqlPart + "::json";
```
Comment (cer) [inline eventuate-common-jdbc/src/main/java/io/eventuate/common/jdbc/sqldialect/PostgresDialect.java:23]: This needs to be conditional on the column type being JSON.

File: mariadb/2.initialize-database.sql [hunk 3]
```diff
@@ -6,8 +6,8 @@ DROP Table IF Exists received_messages;
 CREATE TABLE message (
   id VARCHAR(767) PRIMARY KEY,
   destination LONGTEXT NOT NULL,
-  headers LONGTEXT CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci NOT NULL,
-  payload LONGTEXT CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci NOT NULL,
+  headers JSON NOT NULL,
```
Comment (cer) [inline mariadb/2.initialize-database.sql:9]: JSON should be supported as an **option** - I don't think it's right to make it the default.

File: eventuate-common-jdbc/src/main/java/io/eventuate/common/jdbc/sqldialect/EventuateSqlDialect.java (hunk 1 above)
Comment (dartartem) [reply eventuate-common-jdbc/src/main/java/io/eventuate/common/jdbc/sqldialect/EventuateSqlDialect.java:15]: It is used in cdc, in polling dao: https://github.com/eventuate-foundation/eventuate-cdc/blob/4cda654a694f0cf9a6251a255f7ac76d764d9598/eventuate-local-java-cdc-connector-polling/src/main/java/io/eventuate/local/polling/PollingDao.java#L167

And it is used for string objects. Maybe I will just rename it to stringObjectToString?

File: mariadb/2.initialize-database.sql (hunk 3 above)
Comment (dartartem) [reply mariadb/2.initialize-database.sql:9]: ok, but test cdc is more difficult, but it also should be tested with regular and json columns.
Do you have some thoughts how is better to test it?

File: eventuate-common-jdbc/src/main/java/io/eventuate/common/jdbc/sqldialect/PostgresDialect.java (hunk 2 above)
Comment (dartartem) [reply eventuate-common-jdbc/src/main/java/io/eventuate/common/jdbc/sqldialect/PostgresDialect.java:23]: ok

**Pull request merged**